compiler/noirc_frontend/src/elaborator/comptime.rs

```diff
@@ -399,9 +402,55 @@ impl<'context> Elaborator<'context> {
         }
 
         self.run_attributes_on_functions(functions, &mut generated_items);
+
         generated_items
     }
 
+    fn run_attributes_on_modules(&mut self, generated_items: &mut CollectedItems) {
+        if self.ran_module_attributes {
+            return;
+        }
+        self.ran_module_attributes = true;
```

Comment:
Instead of storing this bool on the elaborator and finding module attributes through the def_maps, I think we should try to add module attributes as another field on `CollectedItems` instead. We don't store anything else for modules there but they can be the first. Since we'll also want to support module-level attributes like `#![foo]` in the future.

Reply:
Done! The code didn't end up as nice as I wanted, though, gathering the attributes happens inside some nested calls instead of in a single place like `collect_functions`. But let me know if there's a way to improve this.

Also, now `ModuleData` ends up having `inner_attributes` and `outer_attributes` but that distinction is no longer needed.... should we just flatten that? For understanding what was going on it helped me to always pass both in functions to know I wasn't forgetting one of them.